base/bucket_test.go

```diff
@@ -231,9 +240,189 @@ func TestGetStatsVbSeqno(t *testing.T) {
+	rootCertPath   = filepath.Join(os.TempDir(), "root.pem")
+	clientKeyPath  = filepath.Join(os.TempDir(), "client.key")
+	clientCertPath = filepath.Join(os.TempDir(), "client.pem")
+)
+
+func mockCertificatesAndKeys(t *testing.T) {
+	notBefore := time.Now().Add(time.Duration(-2) * time.Hour)
+	notAfter := time.Now().Add(time.Duration(2) * time.Hour)
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	assert.NoError(t, err, "Serial number should be generated")
+
+	rootKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	assert.NoError(t, err, "Root key should be generated")
+	saveAsKeyFile(t, rootKeyPath, rootKey)
+
+	rootTemplate := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Organization: []string{"Couchbase, Inc."},
+			CommonName:   "Root CA",
+		},
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+	}
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, &rootTemplate, &rootTemplate, &rootKey.PublicKey, rootKey)
+	assert.NoError(t, err, "Root CA certificate should be generated")
+	saveAsCertFile(t, rootCertPath, certBytes)
+
+	clientTemplate := x509.Certificate{
+		SerialNumber: new(big.Int).SetInt64(4),
+		Subject: pkix.Name{
+			Organization: []string{"Couchbase, Inc."},
+			CommonName:   "client_auth_test_cert",
+		},
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		BasicConstraintsValid: true,
+		IsCA:                  false,
+	}
+
+	clientKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	saveAsKeyFile(t, clientKeyPath, clientKey)
+	certBytes, err = x509.CreateCertificate(rand.Reader, &clientTemplate, &rootTemplate, &clientKey.PublicKey, rootKey)
+	assert.NoError(t, err, "Client certificate should be generated")
+	saveAsCertFile(t, clientCertPath, certBytes)
+}
+
+func saveAsKeyFile(t *testing.T, filename string, key *ecdsa.PrivateKey) {
+	file, err := os.Create(filename)
+	assert.NoError(t, err)
+	defer file.Close()
+	b, err := x509.MarshalECPrivateKey(key)
+	assert.NoError(t, err)
+	err = pem.Encode(file, &pem.Block{Type: "EC PRIVATE KEY", Bytes: b})
+	assert.NoError(t, err)
+}
+
+func saveAsCertFile(t *testing.T, filename string, derBytes []byte) {
+	certOut, err := os.Create(filename)
+	assert.NoError(t, err, "No error while opening cert.pem for writing")
+	err = pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: derBytes})
+	assert.NoError(t, err, "No error while writing data to cert.pem")
+	err = certOut.Close()
+	assert.NoError(t, err, "No error while closing cert.pem")
+}
+
+func deletePaths(paths []string) {
```

Comment:
Calling this deletePaths seems counter-intuitive, since it's not actually calling os.RemoveAll, and doesn't remove the path (just the file).

Reply:
This has been corrected; Renamed this function to removeFiles.